src/raftstore/store/worker/region.rs

```diff
@@ -104,6 +104,39 @@ impl SnapContext {
         let raft_db = self.raft_db.clone();
         let raw_snap = Snapshot::new(self.kv_db.clone());
 
+        let mut old_snaps = None;
```

Comment:
How about `Vec`? An empty `Vec` does not allocate the heap.

Reply:
`None` means we have not try to get all of idle snapshots. `vec![]` means we have poped all items from idle snapshots list. They are different here.

Reply:
please add tests to cover all the conditions here.